mcp_zammad/client.py

```
@@ -481,3 +515,352 @@ def list_tags(self) -> list[dict[str, Any]]:
+            seen.add(cid)
+            visited.append(cid)
+            try:
+                category = self.get_kb_category(kb_id, cid)
+            except ZammadAPIError as exc:
+                if exc.status_code == _HTTP_NOT_FOUND:
+                    logger.warning("KB category %d not found while expanding tree", cid)
+                    continue
+                raise
+            for child_id in category.get("child_ids") or []:
+                if child_id not in seen:
+                    queue.append(child_id)
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)